src/model/ModelObject.cpp

```diff
@@ -97,8 +98,7 @@ namespace model {
       std::vector<IdfObject> removedCosts;
       std::vector<LifeCycleCost> lifeCycleCosts = this->lifeCycleCosts();
       for (LifeCycleCost& lifeCycleCost : lifeCycleCosts) {
-        std::vector<IdfObject> tmp = lifeCycleCost.remove();
-        removedCosts.insert(removedCosts.end(), tmp.begin(), tmp.end());
+        openstudio::detail::concat_helper(removedCosts, lifeCycleCost.remove());
```

Comment:
Some work to try to concat vectors via move operations. see src/utilities/core/ContainersMove.hpp